docs/build.rst

```diff
@@ -0,0 +1,73 @@
+.. _build:
+
+Build Piko/RT
+=============
+
+This chapter will help you start to build ``Piko/RT``, you will need to prepare
+some pacakge and settings.
+
+
+Pre-requirements
+----------------
+
+* svn: used to checkout cmsis.
+* wget: download ``mbed_assert.h`` and ``mbed_preprocessor.h``
+* arm-none-eabi-gcc arm-none-eabi-newlib: cross-compile toolchain.
+
+Arch Linux::
+
+   $ pacman -S base-devel svn lsb-release wget
+   $ pacman -S arm-none-eabi-gcc arm-none-eabi-newlib
+   $ pacman -S python
+
+Ubuntu::
+
+   $ add-apt-repository -y ppa:team-gcc-arm-embedded/ppa
+   $ apt update -qq
+   $ apt install build-essential svn gcc-arm-embedded
+
+MacOS::
+
+   # If you didn't build any thing on MacOS before
+   # please install xcode-select first
+   $ xcode-select --install
+
+   # install brew from https://brew.sh
+   $ brew install subversion wget
+   $ brew tap PX4/homebrew-px4
+   $ brew install gcc-arm-none-eabi-49
```

Comment:
Newest toolchain can download from brew, there isn't any preferable. In my expr, gcc version didn't affect the code. (maybe it have, do the binary need to manage in same version of gcc?)